src/components/QueryEditor.tsx

```diff
@@ -1,56 +1,100 @@
-import React, { ChangeEvent, useEffect, useRef, useState } from 'react';
-import { InlineField, Input, Select } from '@grafana/ui';
+import React, { ChangeEvent, KeyboardEvent, useEffect, useRef, useState } from 'react';
+import { InlineField, Select, TextArea } from '@grafana/ui';
 import { QueryEditorProps, SelectableValue } from '@grafana/data';
 import { CriblDataSource } from '../datasource';
 import { CriblDataSourceOptions, CriblQuery } from '../types';
 import { debounce } from 'lodash';
 
 type Props = QueryEditorProps<CriblDataSource, CriblQuery, CriblDataSourceOptions>;
 
-export function QueryEditor({ datasource, query, onChange, onRunQuery }: Props) {
-  const { maxResults } = query;
+const QUERY_TYPE_OPTIONS = ['saved', 'adhoc'].map((value) => ({ label: value, value }));
+const DEFAULT_QUERY_TYPE = 'adhoc';
 
+export function QueryEditor({ datasource, query, onChange, onRunQuery }: Props) {
   const debouncedOnRunQuery = useRef(debounce(onRunQuery, 750)).current;
   const [savedSearchIdOptions, setSavedSearchIdOptions] = useState<SelectableValue[]>([]);
+  const [savedSearchId, setSavedSearchId] = useState('');
 
-  const onSavedQueryIdChange = (sv: SelectableValue<string>) => {
-    const v = sv.value?.replace(/\s+/g, '') ?? ''; // auto-trim/remove any whitespace
-    if (v.match(/^[a-zA-Z0-9_]+$/)) {
-      onChange({ ...query, savedSearchId: v });
-      debouncedOnRunQuery();
+  const [queryType, setQueryType] = useState(query.type as string ?? DEFAULT_QUERY_TYPE);
```

Comment:
Is `query.type` ever going to exist when the component is initially rendered?

Reply:
I think I was seeing it be present when doing a hard reload of the page.